### DISCUSSION TITLE
Greenfield request: API call to create a hot wallet

### DISCUSSION THREAD

**Discussion by woutersamaey** [btcpayserver/btcpayserver#2596] [General]
As discussed with @Kukks , this is missing from the API and needs more thought.

**Comment by bolatovumar** [comment]
Would this API endpoint also create a store with which the wallet is associated or would we expect the store to have been created by something else first?

**Reply by woutersamaey** [reply]
If the store can be created with acceptable defaults and be fully customizable later, I would not mind for it to be auto-created. 

Would even prefer it really.

Out of the top of my head, I think you'll only need a store name, which could be defaulted to something like "Store for wallet xxx" perhaps?

**Reply by bolatovumar** [reply]
@woutersamaey yeah, only the name is required to create a store. It's kind of weird though that you create a wallet and it create a whole store to go with it as well. I guess it has to be this way with current architecture since wallets are tied to stores. I guess we _could_ create virtual stores which never show up in the UI but exist in the background just to hold a wallet. Not sure if that's a good idea though.

**Reply by woutersamaey** [reply]
Yes it is a bit weird, but we have no other option at this point. I'd set the name to something that doesn't alarm the admin, so "store or wallet xx"" should not look too surprising?

Disconnecting stores and wallets is not coming anytime soon I believe.

**Reply by bolatovumar** [reply]
Yeah, I guess creating _virtual_ stores which are only available through the API as I said above could help a bit as there would be nothing weird in the UI to alarm anyone of anything. Not sure if that's a good idea though.

**Reply by woutersamaey** [reply]
It could be for the time being. If we flag them as `virtual` we'd know they are there and they can be refactored more easily in the future.

**Reply by Kukks** [reply]
This is feature creep for now, let's focus on generating the hot wallet for an existing store (we can do this feature separately as part of  `api/v1/wallets`.) Concept ACK.

**Reply by bolatovumar** [reply]
How would we handle the part where a user receives their seed in the UI and is asked to write it down? Do we just return the string as part of the success response?

**Reply by woutersamaey** [reply]
I believe so, yes. 

I can suggest some alternatives (like needing to decrypt yourself on your end), but that won't make it much safer as the decryption key would also need to be passed the same way.

If you are using HTTPS, you should be fine.
